pkg/pulsar/pulsar.go

```diff
@@ -157,6 +157,71 @@ func (c *Client) Close() error {
 	return nil
 }
 
+type ReceivedSample struct {
+	Sample  *model.Sample
+	Context context.Context
+	Ack     func()
+	Nack    func()
+}
+
+// Receiver watches the queue for relevant samples, unserializes them and sends
+// via the sampleCh. A channel is returned itself, to wait for the work loop to
+// finish
+func (c *Client) Receiver(ctx context.Context, sampleCh chan ReceivedSample) (done chan struct{}, err error) {
+	consumer, err := c.consumer()
+	if err != nil {
+		return nil, err
+	}
+
+	done = make(chan struct{})
+
+	go func() {
+		defer close(done)
+
+		recv := consumer.Chan()
+
+		for {
+			select {
+			case msg := <-recv:
+
+				id := msg.Message.ID()
+				payload := msg.Message.Payload()
+
+				sample, err := c.serializer.Unmarshal(payload)
+				if err != nil {
+					// TODO: Decide if that should be ack or nack
```

Comment:
Seems unlikely that a serialization error would work eventually. I think logging, incrementing a counter, and `ack`-ing is probably the right call

Reply:
Yes agreed, I have put it in partly into c5d1ab8 and the metrics should be implemented as part of #36